src/components/elements/wrapper/types.ts

```diff
@@ -3,12 +3,51 @@ import { IFilterItemSchema } from "../../custom/filter/filter-item/types";
 import { TComponentOmitProps, TFrontendEngineFieldSchema } from "../../frontend-engine";
 import { TRenderRules } from "../../frontend-engine/yup";
 
+type MobileCol = 1 | 2 | 3 | 4;
+type MobileColRange = MobileCol | 5;
+type TabletCol = MobileCol | 5 | 6 | 7 | 8;
+type TabletColRange = TabletCol | 9;
+type DesktopCol = TabletCol | 9 | 10 | 11 | 12;
+type DesktopColRange = DesktopCol | 13;
+export interface ColWrapperProps extends React.HTMLAttributes<HTMLDivElement> {
+	"data-testid"?: string | undefined;
+	/**
+	 * Specifies the number of columns to be span across in mobile viewports.
+	 * If an array is specified, the format is as such [startCol, endCol].
+	 * If `tabletCols` or `desktopCols` are not specified, this
+	 * setting will be applied to tablet and desktop viewports.
+	 *
+	 * If all column props are not specified, the div will span across a single
+	 * column.
+	 */
+	mobile?: MobileCol | [MobileColRange, MobileColRange] | undefined;
+	/**
+	 * Specifies the number of columns to be span across in tablet viewports.
+	 * If an array is specified, the format is as such [startCol, endCol].
+	 * If `desktopCols` are not specified, this setting will be
+	 * applied to desktop viewports as well.
+	 *
+	 * If all column props are not specified, the div will span across a single
+	 * column.
+	 */
+	tablet?: TabletCol | [TabletColRange, TabletColRange] | undefined;
+	/**
+	 * Specifies the number of columns to be span across in desktop viewports.
+	 * If an array is specified, the format is as such [startCol, endCol].
+	 *
+	 * If all column props are not specified, the div will span across a single
+	 * column.
+	 */
+	desktop?: DesktopCol | [DesktopColRange, DesktopColRange] | undefined;
+}
+
```

Comment:
1. as this can be applied to all schemas, move this to `src/components/frontend-engine/types.ts` under a new `COLUMNS` divider

```
// =============================================================================
// JSON SCHEMA
// =============================================================================

// rest of the codes here
```
---

2. `IBaseElementSchema`, `ICustomElementJsonSchema` and `IBaseCustomFieldSchema` should also have the `columns` key

Reply:
add JSDocs comment
```
/** set responsive columns */
```